installation: switch to YAML config files by default (#1551)
fluent/fluent-bit-docs: installation/upgrade-notes.md

The tag drives the overall binary release process so release binaries (containers/packages) will appear after a tag and its associated release note.
This allows users to expect the new release binary to appear and allow/deny/update it as appropriate in their infrastructure.

## Fluent Bit v4.0+

> The default configuration file has been switched to use YAML format and a different name (`.yaml` file suffix instead of `.conf`).

Reviewer: ```suggestion
By default, configuration files use YAML syntax and the `.yaml` file extension. 
```

> This may require updating any custom deployments to either provide a YAML configuration or switch back to using legacy configuration in the arguments to the Fluent Bit binary.

Reviewer: ```suggestion
You may need to update your custom deployments to either use YAML configuration files
or switch back to using legacy configuration in the arguments to the Fluent Bit binary.
```

> For Linux the Systemd unit will need overriding to specify `-c <legacy config file>`.
> Similarly for containers, Windows and macOS the default launch command is changed to point at the YAML config file by default.

Reviewer: ```suggestion
In Linux environments, you must override the systemd unit to specify `-c <legacy config file>`.

Similarly, in Windows and macOS container environments, the default launch command uses YAML configuration files
by default.
```
